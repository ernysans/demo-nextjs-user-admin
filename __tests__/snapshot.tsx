/**
 * @jest-environment jsdom
 */
import {render} from "@testing-library/react";
import Page from '@/app/users/new/page';
import React from 'react';

it("renders users/new/page", () => {
    const {container} = render(<Page/>);
    expect(container).toMatchSnapshot();
});