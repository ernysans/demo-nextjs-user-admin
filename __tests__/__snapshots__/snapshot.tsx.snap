// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders users/new/page 1`] = `
<div>
  <div
    class="space-y-6"
  >
    <h2
      class="text-2xl font-bold mb-4"
    >
      Add New User
    </h2>
    <p
      class="text-gray-600 mb-4"
    >
      Use the form below to add a new user. All fields are required.
    </p>
    <form
      action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')"
      class="mx-auto p-6 bg-white rounded shadow"
    >
      <div
        class="grid grid-cols-1 gap-6"
      >
        <div
          class="grid grid-cols-1 gap-2"
        >
          <label
            class="block text-sm font-medium text-gray-700 mb-1"
            for="name"
          >
            Name
          </label>
          <input
            class="mb-4 block w-full px-3 py-2 border border-gray-300 rounded focus:outline-none focus:ring-2 focus:ring-blue-500"
            id="name"
            name="name"
            required=""
            type="text"
          />
        </div>
        <div
          class="grid grid-cols-1 gap-2"
        >
          <label
            class="block text-sm font-medium text-gray-700 mb-1"
            for="zipCode"
          >
            Zip Code
          </label>
          <input
            class="mb-4 block w-full px-3 py-2 border border-gray-300 rounded focus:outline-none focus:ring-2 focus:ring-blue-500"
            id="zipCode"
            name="zipCode"
            required=""
            type="text"
          />
        </div>
      </div>
      <div
        class="flex justify-between items-center mt-6"
      >
        <a
          class="inline-block px-4 py-2 bg-gray-300 text-gray-800 rounded hover:bg-gray-400 transition border border-gray-400"
          href="/"
        >
          Cancel
        </a>
        <button
          class="px-4 py-2 bg-blue-600 text-white rounded hover:bg-blue-700 transition border border-blue-700 "
          type="submit"
        >
          Add User
        </button>
      </div>
    </form>
  </div>
</div>
`;
